TD_5434 Move the Centre Name to the top of footer across all Learning Portal Pages.

ABSinhaa opened the pull request:
### JIRA link
[TD-5434](https://hee-tis.atlassian.net/browse/TD-5434)

### Description
I added the centre name to the footer and accessed it as a ViewData across the footer partial after making changes to three Learning Hub pages i.e. Current, Available and Completed.

### Screenshots
![image](https://github.com/user-attachments/assets/3658130f-4880-4e55-a789-25f573114a0c)


Wave tool screenshot
![image](https://github.com/user-attachments/assets/871a4c20-b2ab-4225-9f52-3e81312dbbc9)



-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the IDE auto formatter on all files I’ve worked on and made sure there are no IDE errors relating to them
- [ ] Written or updated tests for the changes (accessibility ui tests for views, tests for controller, data services, services, view models created or modified) and made sure all tests are passing
- [x] Manually tested my work with and without JavaScript (adding notes where functionality requires JavaScript)
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related). Addressed any valid accessibility issues and documented any invalid errors
- [x] Updated my Jira ticket with testing notes, including information about other parts of the system that were touched as part of the MR and need  to be tested to ensure nothing is broken
- [ ] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks in the GitHub PR ‘Files Changed’ tab
Either:
- [ ] Documented my work in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3461087233/Development), updating any business rules applied or modified. Updated GitHub readme/documentation for the repository if appropriate. List of documentation links added/changed:
  - [doc_1_here](link_1_here)
Or:
- [ ] Confirmed that none of the work that I have undertaken requires any updates to documentation


[TD-5434]: https://hee-tis.atlassian.net/browse/TD-5434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

kevwhitt-hee reviewed (changes requested):
I have updated the styling and content of the footer to reflect Allan's suggested design.

I don't like the approach of loading the banner text into ViewData and showing if the page name matches a selected list. Feels clunky.

Speaking to Allan, we should show this in the footer wherever the user is logged into a centre. We should look at the approach that we use to load the centre logo into the header (we use a view component) and follow the same pattern with the banner text in the footer.

kevwhitt-hee commented:
Also, following the above, we should remove the banner text from the main body heading in the supervisor pages.

ABSinhaa commented:
@kevwhitt-hee I have added a viewcomponent and then I am loading the banner text into the footer based on the application being selected since we are only doing this for "Learning Portal" and "Supervisor" modules. I have also added a horizontal rule to separate the centre banner content and the links we show below but I have asked Allan for his review on task board, since I had to write a new css class for it as I didn't find any other class that could have done the same like we have on the gov.uk website.

Auldrin-Possa reviewed (commented):
Code looks good however I think need following changes
1. Banner headings/Banner in the footer should not display when banner text is empty. @kevwhitt-hee - Please confirm.

2. In supervisor section, banner text should be displayed from centre to which supervisor is logged-in. Looks like viewdata implementation is missing for supervisor section @kevwhitt-hee - Please confirm.

3. I think implementation is pending for couple of Learning menu pages.
CoursePassword.cshtml
Index.cshtml

kevwhitt-hee commented:
@ABSinhaa, I realise there were some more considerations in this ticket other than just the styling. I think we can use a more generic approach to this and load the banner text into the footer in all parts of the application. If bannertext is null, we should not display the footer.

ABSinhaa commented:
@kevwhitt-hee I have made the changes that you requested in the last comment, this is now an application wide change which will only show itself if a BannerText value exists, I have further made changes to the Learning Menu pages as directed by @Auldrin-Possa in his earlier comments. Moreover, the 'NEED HELP ?' text would only show if there's a banner text value assigned. Kindly let me know if there's anything else that I can do regarding this task.

kevwhitt-hee reviewed (changes requested):
Please could we change the way that we are loading the banner text so that it is loaded into the footer throughout the application independent of the page in which it is being displayed (not using the page's VieWData)? Please see the implementation of the centre logo in the header for a model approach.

ABSinhaa commented:
@kevwhitt-hee and @rshrirohit  I have made the changes that we discussed over this morning, kindly take a look when you get some time. Also performed a clean up wherever I found the changes views injecting the banner text. 
 
 Thank You.
